### PR TITLE
Add functionality to request versions list

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Implement get versions list in CatalogRepository.
Add tests for CatalogRepository.

Relates-To: OLPEDGE-1606

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp [hunk 1]
```diff
@@ -152,6 +153,35 @@ CatalogVersionResponse CatalogRepository::GetLatestVersion(
   return version_response;
 }
 
+VersionsResponse CatalogRepository::GetVersionsList(
+    const client::HRN& catalog,
+    client::CancellationContext cancellation_context,
+    const VersionsRequest& request, const client::OlpClientSettings& settings) {
+  auto fetch_option = request.GetFetchOption();
+  repository::CatalogCacheRepository repository(catalog, settings.cache);
+  if (fetch_option == CacheOnly) {
```
Comment (meravingen123) [inline olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp:162]: What about CacheWithUpdate option?

Comment (LiubovDidkivska) [reply olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp:162]: It is not clear for now how to cache requests. So it is temporary will request online.

Comment (andescu) [reply olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp:162]: CacheWithUpdate does not make much sense for versioned catalogs as you always get the same result back. It is more for volatile layers as there the value can change every time you request.

File: olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp [hunk 2]
```diff
@@ -28,35 +28,47 @@
...
 
-#define OLP_SDK_URL_LOOKUP_CONFIG \
-  R"(https://api-lookup.data.api.platform.here.com/lookup/v1/platform/apis)"
+constexpr auto kUrlLookupConfig =
+    R"(https://api-lookup.data.api.platform.here.com/lookup/v1/platform/apis)";
 
-#define OLP_SDK_HTTP_RESPONSE_LOOKUP_CONFIG \
-  R"jsonString([{"api":"config","version":"v1","baseURL":"https://config.data.api.platform.in.here.com/config/v1","parameters":{}},{"api":"pipelines","version":"v1","baseURL":"https://pipelines.api.platform.in.here.com/pipeline-service","parameters":{}},{"api":"pipelines","version":"v2","baseURL":"https://pipelines.api.platform.in.here.com/pipeline-service","parameters":{}}])jsonString"
+constexpr auto kResponseLookupConfig =
+    R"jsonString([{"api":"config","version":"v1","baseURL":"https://config.data.api.platform.in.here.com/config/v1","parameters":{}},{"api":"pipelines","version":"v1","baseURL":"https://pipelines.api.platform.in.here.com/pipeline-service","parameters":{}},{"api":"pipelines","version":"v2","baseURL":"https://pipelines.api.platform.in.here.com/pipeline-service","parameters":{}}])jsonString";
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp:55]: Remove empty new line.

File: olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp [hunk 3]
```diff
@@ -28,35 +28,47 @@
...
+constexpr auto
+    kResponseConfig =
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp:48]: Please format.

File: olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp [hunk 4]
```diff
@@ -72,13 +84,16 @@ const std::string kConfigCacheKey =
 const std::string kLookupUrl =
     "https://api-lookup.data.api.platform.here.com/lookup/v1/resources/" +
     kCatalog + "/apis/" + kMetadataServiceName + "/" + kServiceVersion;
+const std::string kVersionInfosCacheKey = kCatalog + "::3::4::versionInfos";
+const std::string kVersionInfosMinCacheKey = kCatalog + "::-1::4::versionInfos";
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp:88]: Used?

File: olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp [hunk 5]
```diff
@@ -562,4 +588,84 @@ TEST_F(CatalogRepositoryTest, GetCatalogTimeout) {
   EXPECT_EQ(olp::client::ErrorCode::RequestTimeout,
             response.GetError().GetErrorCode());
 }
+
+TEST_F(CatalogRepositoryTest, GetVersionsList) {
+  {
+    SCOPED_TRACE("Get versions list");
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp:594]: Were is the test for min_version = -1?

File: olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp (hunk 3 above)
Comment (LiubovDidkivska) [reply olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp:48]: Do you mean that it begins from new line? It is actually how clang formats it.

**Pull request merged**